testbed/tests/app/test_desktop.py

```diff
@@ -201,19 +203,20 @@ async def test_presentation_mode(app, app_probe, main_window, main_window_probe)
         screen_window_dict[window_information["paired_screen"]] = window_information[
             "window"
         ]
-
-    # Add delay to ensure windows are visible after animation.
+    # Wait for window animation before assertion.
     await main_window_probe.wait_for_window("All Test Windows are visible")
 
     # Enter presentation mode with a screen-window dict via the app
     app.enter_presentation_mode(screen_window_dict)
     # Add delay to ensure windows are visible after animation.
-    await main_window_probe.wait_for_window(
-        "App is in presentation mode", full_screen=True
-    )
-    assert app.in_presentation_mode
+    await main_window_probe.wait_for_window("App is in presentation mode")
     # All the windows should be in presentation mode.
     for window_information in window_information_list:
+        # Wait for window animation before assertion.
+        await window_information["window_probe"].wait_for_window(
+            "App is in presentation mode", state=WindowState.PRESENTATION
+        )
+        assert app.in_presentation_mode
```

Comment:
This assertion only needs to be executed once, not once per window. It needs to be after we've waited for all windows, so putting it after the for loop would make sense to me.

Reply:
I have put the assertion after the for loop.